sdv/multi_table/utils.py

```diff
@@ -621,20 +608,11 @@ def _subsample_data(data, metadata, main_table_name, num_rows):
     result = deepcopy(data)
     primary_keys_referenced = _get_primary_keys_referenced(result, metadata)
     ratio_to_keep = num_rows / len(result[main_table_name])
-    try:
-        _validate_foreign_keys_not_null(metadata, result)
-    except SynthesizerInputError:
-        warnings.warn(
-            'The data contains null values in foreign key columns. '
-            'We recommend using ``drop_unknown_foreign_keys`` method from sdv.utils'
-            ' to drop these rows before using ``get_random_subset``.'
-        )
 
     try:
         _subsample_disconnected_roots(result, metadata, main_table_name, ratio_to_keep)
         _subsample_table_and_descendants(result, metadata, main_table_name, num_rows)
         _subsample_ancestors(result, metadata, main_table_name, primary_keys_referenced)
```

Comment:
Rather than removing the the code, should be make it controllable via a parameter on `_subsample_data`?

Reply:
Alright, done in [a4f8295](https://github.com/sdv-dev/SDV/pull/2082/commits/a4f8295ea9757de1b64ad35506cd71080005ae82)